src/plugins/event-plugins/WebVitalsPlugin.ts

```diff
@@ -34,28 +47,66 @@ export class WebVitalsPlugin extends InternalPlugin {
     configure(config: any): void {}
 
     protected onload(): void {
+        this.context.eventBus.subscribe(Topic.EVENT, this.lcpHelper); // eslint-disable-line
         onLCP((metric) => this.handleLCP(metric));
         onFID((metric) => this.handleFID(metric));
         onCLS((metric) => this.handleCLS(metric));
     }
 
-    handleLCP(metric: LCPMetricWithAttribution | Metric) {
+    private lcpHelper(msg: any) {
+        const event = msg as ParsedRumEvent;
+        switch (event.type) {
+            // lcp resource is either image or text
+            case PERFORMANCE_RESOURCE_EVENT_TYPE:
+                const details = event.details as ResourceEvent;
+                if (
+                    details.fileType === ResourceType.IMAGE ||
+                    details.initiatorType === 'img'
+                ) {
+                    this.resourceEventIds.set(
+                        performanceKey(event.details as HasLatency),
+                        event.id
+                    );
+                }
+                break;
+            case PERFORMANCE_NAVIGATION_EVENT_TYPE:
+                if (!this.navigationEventId) {
+                    this.navigationEventId = event.id;
+                }
```

Comment:
```suggestion
                this.navigationEventId = event.id;
```

Reply:
I wanted to make this less brittle in case multiple navigation events are captured in the future, but I guess that's not a thing that will happen.